### PR TITLE
UCP/TEST: Add utility to pack/unpack addresses.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1036/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1648/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1038/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1650/ for details.

File: src/ucp/wireup/address.c [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+}
+
+static ucp_address_packed_device_t*
+ucp_address_get_device(const char *name, ucp_address_packed_device_t *devices,
+                       ucp_rsc_index_t *num_devices_p)
+{
+    ucp_address_packed_device_t *dev;
+
+    for (dev = devices; dev < devices + *num_devices_p; ++dev) {
+        if (!strcmp(name, dev->dev_name)) {
```
Comment (miked-mellanox) [inline src/ucp/wireup/address.c]: unsafe compare, assumes name and dev_name same len

Comment (yosefe) [inline src/ucp/wireup/address.c]: There is no such assumption, strcmp takes into account string length

Comment (miked-mellanox) [inline src/ucp/wireup/address.c]: ohh. right. thx

**Comment by miked-mellanox** [comment]
:+1:

**Comment by yosefe** [comment]
@shamisp can you pls review ?

**Comment by shamisp** [comment]
:+1:

**Pull request merged**